### PR TITLE
[WIP] Introducing 'license' field for entries

### PR DESCRIPTION
**Pull request by merkys**
Fixes (partially) #102 by introducing per-entry field to specify licensing details.

### CODE REVIEW

File: optimade.md [hunk 1]
```diff
@@ -1332,6 +1333,25 @@ This section defines standard entry types and their properties.
   * \_exmpl\_trajectory
   * \_exmpl\_workflow\_id
 
+### <a name="h.6.1.4">6.1.4. license</a>
+
+* **Description**: Dictionary describing the license of the entry.
+* **Requirements/Conventions**: MUST contain one or both of the following fields:
```
Comment (CasperWA) [inline optimade.md]: (MUST) be a dictionary containing (~contain~ one or both ...)

Comment (CasperWA) [inline optimade.md]: Perhaps state whether this is a mandatory field or optional?

**Comment by giovannipizzi** [comment]
Marked as WIP because we are having some discussions.
It's not every single structure that is under license, but the license applies to the DB, i.e., to the work that puts together many structures bringing additional knowledge. The single structure, even if it comes from a proprietary DB, is still under the license of the original paper.

File: optimade.md (hunk 1 above)
Comment (merkys) [reply optimade.md]: Fixed.

Comment (merkys) [reply optimade.md]: This seems to be a subject of discussion. Nevertheless, the requirement level might be added via #2.

**Comment by rartino** [comment]
This seems quite dangerous. 

If we allow every single data entry in a database to carry a license of its own, then---separate from the consideration of whether it really is possible for entries to be individually licensed---it would fall onto users to verify every single data entry to see whether further processing breaks its license terms. One may receive a single entry with a more restrictive license hidden among millions of "openly" licensed entries.

Hence, it makes more sense to only allow a single per-base_url license info. That license info can, however, provide a url that explains a more complex licensing situation that applies when, e.g., two databases have been merged.

I do see the issue for a sky-scanner type service. However, such a service can explain in its license field that it returns structures out of multiple databases and that the license depend on the data from which the data is fetched. Then it would add references relations to everything it returns that point to the source. (Which it should be doing anyway.)

**Comment by rartino** [comment]
So, how do we want to go about with this one? Should it be closed as "we won't do this". Or is @merkys willing to re-edit it to instead allow a license on the database as a whole?

My wishlist for a database-wide license field:
- Some standard way of saying "its complicated" and a URL. 
- Some standard way of saying "all our data may be further distributed via downstream OPTiMaDe APIs." to allow a skyscanner-type implementation to automatically detect whether it is allowed to redistribute data.

**Comment by merkys** [comment]
> So, how do we want to go about with this one? Should it be closed as "we won't do this". Or is @merkys willing to re-edit it to instead allow a license on the database as a whole?

I am closing this PR. Per-database license is a different thing. I will open another PR once it's ready.

**Pull request closed without merging**